### DISCUSSION TITLE
Repeat Texture/Material

### DISCUSSION THREAD

**Discussion by kybarg** [bevyengine/bevy#6893] [Q&A]
Maybe stupid quitsion, but haven't found any working solution.

So I have a **bricks material**, and **plane mesh**.
When I apply material it is rendered texture is stretched acrosll all the mesh.
I want **texture** to be **repeated** 10 times instead of stretching.

**Comment by torsteingrindvik** [comment]
Hi, if I understand you correctly you want to use repeat mode on the texture.
To my knowledge you can't specify which sampler address mode you want to use when you load an image in Bevy right now.

My solution was to store the handle of the image I want to change, then switch its sampler when it loads.
I've done that here: https://github.com/torsteingrindvik/bevy-vfx-bag/blob/debc1a5af06398c7e62e4efac3d9eb8b3fcc836a/src/image/raindrops.rs

**Comment by janhohenheim** [comment]
I've resorted to using a [shader](https://github.com/janhohenheim/foxtrot/blob/main/assets/shaders/repeated.wgsl) controller by [this material](https://github.com/janhohenheim/foxtrot/blob/v0.1.5/src/shader.rs#L65). Most of the shader is just boilerplate copied from the default PBR shader to not lose light and shadow support.
